app/controllers/exercises_controller.rb

```diff
@@ -58,26 +58,31 @@ def duplicate
 
   # GET /exercises/1/edit
   def edit
+    exercise_relation = ExerciseRelation.find_by(clone_id: params[:id])
+    if exercise_relation
+      @exercise_relation = exercise_relation
+    end
   end
 
   # POST /exercises
   # POST /exercises.json
   def create
     @exercise = Exercise.new(exercise_params)
-    @exercise.add_attributes(params[:exercise])
     @exercise.user = current_user
-    exercise_dependencies
+    clone = false
+    clone = true if params[:exercise][:exercise_relation]
 
```

Comment:
this is a little strange to read. I'd rather go for 

if ....
clone = true
else 
clone = false
end